astropy/coordinates/tests/test_sky_coord.py

```diff
@@ -1542,6 +1522,35 @@ def test_spherical_offsets():
     assert_allclose(dra, 0*u.arcmin)
     assert_allclose(ddec, np.arange(4)*u.arcmin)
 
+    # spherical_offsets_by only supports scalar frame data:
+    with pytest.raises(ValueError):
+        i01s.spherical_offsets_by(-dra, -ddec)
+
+
+@pytest.mark.parametrize('frame', ['icrs', 'galactic'])
+@pytest.mark.parametrize('comparison_data', [(0*u.arcmin, 1*u.arcmin),
+                                             (1*u.arcmin, 0*u.arcmin),
+                                             (1*u.arcmin, 1*u.arcmin)])
+def test_spherical_offsets_roundtrip(frame, comparison_data):
+    i00 = SkyCoord(0*u.arcmin, 0*u.arcmin, frame=frame)
+    comparison = SkyCoord(*comparison_data, frame=frame)
+
+    dlon, dlat = i00.spherical_offsets_to(comparison)
+    assert_allclose(dlon, comparison.data.lon)
+    assert_allclose(dlat, comparison.data.lat)
+
+    i00_back = comparison.spherical_offsets_by(-dlon, -dlat)
+    # TODO: this roundtripping accuracy isn't great
```

Comment:
Is that unexpected?

Reply:
Hm, I would have expected better precision but maybe just haven't thought through it...

Reply:
The offsets in RA when dec is changing too should differ by a factor cos(dec+theta)/cos(dec)); for dec=0, this implies a difference of ~ 1-cos(theta) ~ 0.5theta^2 ~ 4e-8, so I'd have expected 4e-8/60=7e-10 deg. That seems about OK...
Maybe add to comment? Perhaps even to docstring?

Reply:
Feeling dense today so I don't follow 😅  - shouldn't the accuracy be better than that, because the rotations should be exact? As in, shouldn't the inverse of `spherical_offsets_to()` be an exact transformation? Here's the accuracy I'm getting:
```python
frame = 'icrs'
i00 = SkyCoord(0*u.arcmin, 0*u.arcmin, frame=frame)

for comparison_data in [(0*u.arcmin, 1*u.arcmin),
                        (1*u.arcmin, 0*u.arcmin),
                        (2*u.arcmin, 2*u.arcmin)]:
    comparison = SkyCoord(*comparison_data, frame=frame)

    dlon, dlat = i00.spherical_offsets_to(comparison)
    i00_back = comparison.spherical_offsets_by(-dlon, -dlat)
    
    mhvk_factor = np.cos(comparison.data.lat + -dlat) / np.cos(comparison.data.lat)
    
    print("{:.3e} {:.3e} {:.3e}".format(
        (i00.data.lon - i00_back.data.lon).to_value(u.rad), 
        (i00.data.lat - i00_back.data.lat).to_value(u.rad),
        1 - mhvk_factor))
```

Outputs:
```
0.000e+00 0.000e+00 -4.231e-08
-2.645e-16 0.000e+00 0.000e+00
-9.845e-11 9.846e-11 -1.692e-07
```

Reply:
My argument definitely was only the case of offsets in both dec and ra. Plus, it is the *relative* error, so it should be `(1-mhvk_factor)*offset = 1.69e-7*2*pi/(180*60)=1e-10`

Reply:
OK I added a note as a comment in the test